src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -190,6 +196,25 @@ private void InvalidateOrUpdateBySyntaxCheckDelimeters(ManagedList<Diagnostic> d
             finally { diagnostics.SyncRoot.ExitWriteLock(); }
         }
 
+        /// <summary>
+        /// Replaces the given list of latest diagnostics with a copy of the current ones, 
+        /// Updates the line numbers for both the current and latest diagnostics that start after the syntax check end delimiter, 
+        /// and removes all diagnostics that overlap with the given range for teh syntax check update in the latest diagnostics only. 
+        /// Throws an ArgumentNullException if the given current and/or latest diagnostics are null, or if the syntax check delimiters are null. 
+        /// Throws an ArgumentException if the given start and end position do not denote a valid range.
+        /// </summary>
+        private static void DelayInvalidateOrUpdate(ManagedList<Diagnostic> current, ManagedList<Diagnostic> latest,
+            Range syntaxCheckDelimiters, int lineNrChange)
+        {
+            if (current == null) throw new ArgumentNullException(nameof(current));
+            if (latest == null) throw new ArgumentNullException(nameof(latest));
+
+            latest.ReplaceAll(current);
```

Comment:
What is current and latest here? Is latest newer than current - but it's being replaced by current?

Reply:
Yes, the latest is used to store partially computed diagnostics and thus always newer than latest. The replacement was indeed an artifact of an early draft - I removed it.

Reply:
Another name for "latest" like "next" might help. It could just be me but I am a little confused by the difference between "current" and "latest". :)

Reply:
renamed to diagnostics and updated